docs/usage/dto/1-dto-factory.rst

```diff
@@ -268,3 +268,212 @@ This can be overridden by setting the
     :language: python
     :linenos:
     :emphasize-lines: 14,15
+
+Enveloping Return Data
+----------------------
+
+Litestar's DTO Factory types are versatile enough to manage your data, even when it's nested within generic wrappers.
+
+The following example demonstrates a route handler that returns DTO managed data wrapped in an enveloping type. The
+envelope is used to deliver additional metadata about the response - in this case, a count of the number of items
+returned. Read on for an explanation of how this works.
+
+.. literalinclude:: /examples/data_transfer_objects/factory/enveloping_return_data.py
+    :caption: Enveloping Return Data
+    :language: python
+    :linenos:
+
+First, create a generic dataclass to act as your envelope. This envelope will contain your data and any additional
+attributes you might need. In this example, we have a ``CountEnvelope`` dataclass which has a ``count`` attribute.
+The envelope must be a python generic type with one or more type parameters, and at least one of those type parameters
+should describe an instance attribute that will be populated with the data.
+
+.. code-block:: python
+
+   from dataclasses import dataclass
+   from typing import Generic, TypeVar
+
+   T = TypeVar("T")
+
+
+   @dataclass
+   class CountEnvelope(Generic[T]):
+       count: int
+       data: T
+
+
+Now, create a DTO for your data object and configure it using ``DTOConfig``. In this example, we're excluding
+``password`` and ``created_at`` from the final output.
+
+.. code-block:: python
+
+   from litestar.contrib.sqlalchemy.dto import SQLAlchemyDTO
+   from litestar.dto.factory import DTOConfig
+
+
+   class UserDTO(SQLAlchemyDTO[User]):
+       config = DTOConfig(exclude={"password", "created_at"})
+
+Then, set up your route handler. This example sets up a ``/users`` endpoint, where a list of ``User`` objects is
+returned, wrapped in the ``CountEnvelope`` dataclass.
+
+.. code-block:: python
+
+   from litestar import get
+
+
+   @get("/users", dto=UserDTO, sync_to_thread=False)
+   def create_user() -> CountEnvelope[List[User]]:
+       return CountEnvelope(
+           count=1,
+           data=[
+               User(
+                   id=1,
+                   name="Litestar User",
+                   password="xyz",
+                   created_at=datetime.now(),
+               ),
+           ],
+       )
+
+
+This setup allows the DTO to manage the rendering of ``User`` objects into the response. The DTO Factory type will find
+the attribute on the enveloping type that holds the data and perform its serialization operations upon it.
+
+Returning enveloped data is subject to the following constraints:
+
+#. The type returned from the handler must be a type that litestar can natively encode.
```

Comment:
```suggestion
#. The type returned from the handler must be a type that Litestar can natively encode.
```

Reply:
sorry i didnt find anything else 🙃  maybe this is good thing

Reply:
> maybe this is good thing

I'm definitely going to interpret this as such :)